papergrid/src/lib.rs

```diff
@@ -396,7 +396,31 @@ impl Grid {
         }
     }
 
-    pub fn extract<R, C>(&self, rows: R, columns: C) -> Self
+    /// Returns a new [Grid] that reflects a segment of the referenced [Grid]
+    ///
+    /// The segment is defined by [RangeBounds<usize>] for Rows and Columns
+    ///
+    /// # Example
+    ///
+    /// ```rust,no_run
+    /// grid
+    /// +---+---+---+
+    /// |0-0|0-1|0-2|
+    /// +---+---+---+
+    /// |1-0|1-1|1-2|
+    /// +---+---+---+
+    /// |2-0|2-1|2-2|
+    /// +---+---+---+
+    /// grid.extract(.., ..1)
+    /// +---+
+    /// |0-0|
+    /// +---+
+    /// |1-0|
+    /// +---+
+    /// |2-0|
+    /// +---+
+    /// ```
+    pub fn extract<R, C>(&self, rows: &R, columns: &C) -> Self
```

Comment:
I may missed something but why these references are so  important?

```rust
rows: &R, columns: &C
```

Reply:
I could use some help on this issue. 
I made these changes to the `papergrid` extract api because I was unable to dereference the provided `RangeBounds` input from the `tabled` extract api.

I think it is super valuable to allow a user to write something like the following in the api of the exposed package.

```rust
Extract::new(.., 1..)
```

[TableOption change function](https://github.com/zhiburt/tabled/pull/87/files#diff-f3fcb4eeee8bd925542056f5ea55be80ef332b1c41fd746b30e0fc255813e8bfR68-R76)

I also see that you are suggesting to change the Extract api implementation to use functions over the New/TableOption api! That very well could address this issue entirely. Good idea I think

Reply:
Unfortunately, I have been unable to resolve this issue. 

The comment above explains the issue I'm still having.